heron/metricscachemgr/tests/java/com/twitter/heron/metricscachemgr/metricscache/CacheCoreTestTreeIndex.java

```diff
@@ -0,0 +1,223 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.twitter.heron.metricscachemgr.metricscache;
+
+
+import java.util.Comparator;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsFilter;
+
+import static com.twitter.heron.metricscachemgr.metricscache.MetricsCacheQueryUtils.Granularity.RAW;
+
+public class CacheCoreTestTreeIndex {
+  // test target
+  private static CacheCore cacheCore;
+  // cache timestamp
+  private static long now;
+  // aggregation type
+  private static MetricsFilter metricsFilter;
+  // sort MetricTimeRangeValue
+  private static Comparator<MetricsCacheQueryUtils.MetricTimeRangeValue> comparator;
+
+  @BeforeClass
+  public static void insertData() {
```

Comment:
By consolidating these tests into one class you don't need to copy the insertData method in multiple classes.